CovertInfiltration/Src/CovertInfiltration/Classes/X2EventListener_Infiltration.uc

```diff
@@ -41,6 +46,107 @@ static protected function EventListenerReturn NumCovertActionToAdd(Object EventD
 	return ELR_NoInterrupt;
 }
 
+static protected function EventListenerReturn UpdateReturningSoldiers(Object EventData, Object EventSource, XComGameState GameState, Name EventID, Object CallbackData)
+{
+	local XComGameState_CovertAction CovertAction;
+
+	CovertAction = XComGameState_CovertAction(EventSource);
+
+	if (CovertAction == none)
+	{
+		return ELR_NoInterrupt;
+	}
+
+	UpdateSquadWill(CovertAction, GameState);
+	
+	return ELR_NoInterrupt;
+}
+
+static function UpdateSquadWill(XComGameState_CovertAction CovertAction, XComGameState GameState)
```

Comment:
`GameState` -> `NewGameState` to make it clearly visible that it's a gamestate in progress